cds/modules/deposit/indexer.py

```diff
@@ -43,3 +47,52 @@ def cdsdeposit_indexer_receiver(
     if record['$schema'] in [project_schema, video_schema]:
         json['_cds']['state'] = deposit['_cds']['state']
         json['_files'] = deposit['_files']
+
+
+class CDSRecordIndexer(RecordIndexer):
+
+    def index(self, deposit, action='commit'):
+        project_schema = current_jsonschemas.path_to_url(Project._schema)
+        if action == 'publish':
+            if deposit['$schema'] == project_schema:
+                # index videos (records)
+                pid_values = Project(data=deposit).video_ids
+                ids = [
+                    str(p.object_uuid)
+                    for p in PersistentIdentifier.query.filter(
+                        PersistentIdentifier.pid_value.in_(pid_values)).all()
+                ]
+                # index project (record)
+                _, record = deposit.fetch_published()
+                ids.append(str(record.id))
+                # index deposit (deposit)
+                # FIXME index also deposit videos?
```

Comment:
Probably not needed, because handled by the `commit` action below, no?
A `commit` happens every time we save, if I am not wrong.

Reply:
This `commit` action is not related to the record.commit or session.commit. Is just a name in order to have some default behaviour if we do not pass an action. For the FIXME the thing is that now we are not searching on deposit videos but only the project, that's why is not needed for now, but we can easily do that.